torch/csrc/serialization.cpp

```diff
@@ -97,19 +104,75 @@ static inline ssize_t doPythonIO(PyObject* fildes, void* buf, size_t nbytes, boo
   // fildes.readinto can return UnsupportedOperation so fall back to fildes.read.
   if (is_read && isUnsupportedOperation()) {
     PyErr_Clear();
-    return doPythonReadBuffered(fildes, buf, nbytes);
+    return doPartialPythonReadBuffered(fildes, buf, nbytes);
   }
   throw python_error();
 }
 
 // Call Python fildes.readinto(buf)
-static ssize_t doPythonReadInto(PyObject* fildes, void* buf, size_t nbytes) {
-  return doPythonIO(fildes, buf, nbytes, /* is_read */ true);
+static ssize_t doPartialPythonReadInto(PyObject* fildes, void* buf, size_t nbytes) {
+  return doPartialPythonIO(fildes, buf, nbytes, /* is_read */ true);
 }
 
 // Call Python fildes.write(buf)
-static ssize_t doPythonWrite(PyObject* fildes, void* buf, size_t nbytes) {
-  return doPythonIO(fildes, buf, nbytes, /* is_read */ false);
+static ssize_t doPartialPythonWrite(PyObject* fildes, void* buf, size_t nbytes) {
+  return doPartialPythonIO(fildes, buf, nbytes, /* is_read */ false);
+}
+
+// Requires that we read EXACTLY nbytes; fails if we don't.
+template <typename io>
+void doRead(io fildes, void* raw_buf, size_t nbytes) {
+  char* buf = static_cast<char*>(raw_buf);
+  while (nbytes > 0) {
+    errno = 0; // doPartialRead may not set errno
+    // we read in 1GB blocks to avoid bugs on some OSes
```

Comment:
I'm curious. What are the bugs?

Reply:
The line appears to track back to #1031... and now that I know about I had better cite it here....